internal/engine/buildcontroller.go

```diff
@@ -234,16 +229,11 @@ func buildStateSet(ctx context.Context, manifest model.Manifest, specs []model.T
 		result[id] = buildState
 	}
 
-	// If there are no files changed across the entire state set, then this is a force update.
-	// We want to do an image build of each image.
-	// TODO(maia): I think that instead of storing this on every build state, we can can figure
-	//  out that it's a force update when creating the BuildEntry (in `needsBuild`), store that
-	//  as the BuildReason, and pass the whole BuildEntry to the builder (so the builder can
-	//  know whether to skip in-place builds)
-	// if we're on a crash rebuild, then there won't have been any files changed for that reason
-	if !ms.NeedsRebuildFromCrash && !anyFilesChangedSinceLastBuild {
+	// Check to see if this is a resource in auto mode with an overriding trigger.
+	// See comments on IsOverrideTrigger for more details on the intended flow.
+	if manifest.TriggerMode == model.TriggerModeAuto && ms.TriggerReason != 0 {
 		for k, v := range result {
-			result[k] = v.WithNeedsForceUpdate(true)
+			result[k] = v.WithIsOverrideTrigger(true)
```

Comment:
it feels a little weird with the model to me to store this information on each individual target's state, doesn't it apply to the BuildStateSet as a whole? (I'm not sure it matters enough to change tho--but fwiw storing this info per-target was a DB hack that never got cleaned up)